docs/source/release_notes/release-notes.rst

```diff
@@ -50,16 +50,29 @@ develop
     *    - Type
          - Change
 
-    *    - |devbreak| |improved|
-         - The `TableMetadata` class has been refactored to use Immutables.
-           (`Pull Request <https://github.com/palantir/atlasdb/pull/3624>`__)
+    *    - |new|
```

Comment:
Maybe note that this is now a beta feature?

Reply:
Added an explicit mention, do let me know if you think the wording should be stronger.